### PR TITLE
fixes dynamic querying of transactions by status in the loop while updating the status

### PR DESCRIPTION
**Pull request by saipraveen18**
https://www.pivotaltracker.com/n/projects/2640059/stories/186527854

### CODE REVIEW

File: spec/operations/fdsh/h41/transmissions/batch_request_director_spec.rb [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    DatabaseCleaner.clean
+  end
+
+  let(:report_kind) { :h41_1095a }
+
+  context 'with invalid input params' do
+    context 'bad report_type' do
+      let(:input_params) do
+        {}
+      end
```
Comment (saikumar9) [inline spec/operations/fdsh/h41/transmissions/batch_request_director_spec.rb:26]:
```suggestion
      let(:input_params) { {} }
```

File: spec/operations/fdsh/h41/transmissions/batch_request_director_spec.rb [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        expect(result.failure).to eq('transactions collection required')
+      end
+    end
+
+    context 'bad report_type' do
+      let(:input_params) do
+        {
+          transactions: [double]
+        }
+      end
```
Comment (saikumar9) [inline spec/operations/fdsh/h41/transmissions/batch_request_director_spec.rb:39]:
```suggestion
      let(:input_params) { { transactions: [double] } }
```

**Review by saikumar9 — APPROVED** [review]
lgtm! Good Catch 👏

**Review by vkghub — APPROVED** [review]
LGTM

**Pull request merged**